addons/http_routing/tests/test_lang_url.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch
+from odoo.tests import HttpCase
+
+
+class TestLangUrl(HttpCase):
+    post_install = True
+    at_install = False
+
+    # Mock to simulate hr_HR lang is activated
+    def _get_language_codes(self):
+        return [
+            ('en_US', 'English'),
+            ('hr_HR', 'Croatian / hrvatski jezik'),
+        ]
+
+    # Mock to simulate hr_holidays module is installed
+    def _installed(self):
+        return {
+            module.name: module.id
+            for module in self.env['ir.module.module'].sudo().search(['|', ('state', '=', 'installed'), ('name', '=', 'hr_holidays')])
+        }
+
+    def test_01_url_module_name(self):
+        patcher = patch('odoo.addons.http_routing.models.ir_http.IrHttp._get_language_codes', wraps=self._get_language_codes)
+        patcher.start()
+        self.addCleanup(patcher.stop)
+        patcher2 = patch('odoo.addons.base.module.module.Module._installed', wraps=self._installed)
+        patcher2.start()
+        self.addCleanup(patcher2.stop)
+
+        # Do not remove this `website` fix, see commit message
+        patcher3 = patch('odoo.addons.website.models.ir_http.Http._get_language_codes', wraps=self._get_language_codes)
```

Comment:
I don't like this. It does work without checking if website is installed as the method exists in file but.. not very Odoo friendly as this is not 'modulaire'.

Problem is, this need to be tested in `http_routing`, I don't want to test it in `website`.

But `website` redefine `_get_language_codes` so this test works fine if `website` is not installed, but once it is installed this test stop working as it goes through the method in website module which is not mocked and do not return `Croatian`.

I don't see any better way but if you have any idea I'll take it.

Reply:
Oh and obviously I can't mock `get_nearest_lang` which is the one calling the mocked  `_get_language_codes`. Otherwise the test would be pointless

Reply:
This is not relevant as I don't patch `website` anymore